src/components/FoodDetailsSummary/FoodDetailsSummary.tsx

```diff
@@ -1,45 +1,64 @@
 import React from "react";
 import "react-circular-progressbar/dist/styles.css";
 import { CircularProgressbarWithChildren, buildStyles } from "react-circular-progressbar";
+import { useStore } from "../../helpers/stores";
+import { calculateCaloriesIntake } from "../../helpers";
 
 const FoodDetailsSummary = () => {
-  const nutritionSummaryItems = [
-    { percentage: 25, weight: 200, name: "Calories" },
-    { percentage: 23, weight: 76, name: "Carbs", unit: "g" },
-    { percentage: 12, weight: 1, name: "Protein", unit: "g" },
-    { percentage: 45, weight: 40, name: "Fat", unit: "g" },
-  ];
+  const { nutrition } = useStore((state) => state);
+  const sortedNutrients = nutrition?.slice()?.sort((a, b) => a.displayOrder - b.displayOrder);
   return (
-    <div className="w-full overflow-x-auto scrollbar-none px-[16px] bg-white -mt-4 custom-scrollbar">
-      <h1 className="prose-titleMedium text-textDark font-500 mt-5 mb-4">SUMMARY</h1>
-      <div className="flex justify-between">
-        {nutritionSummaryItems.map((item, idx) => (
-          <div key={idx}>
-            <div className="mt-2 flex justify-center items-center flex-col">
-              <div className="w-[50px] h-[50px]">
-                <CircularProgressbarWithChildren
-                  strokeWidth={10}
-                  value={item.percentage}
-                  styles={buildStyles({
-                    textColor: "#000",
-                    pathColor: "#6660FF",
-                    trailColor: "#E2DFFF",
-                    textSize: "16px",
-                  })}
-                >
-                  <p className="text-center text-[12px] text-main leading-[28px] font-500">
-                    {item.percentage}%
-                  </p>
-                </CircularProgressbarWithChildren>
+    <div className=" px-[16px] bg-white -mt-4">
+      <h1 className="prose-titleMedium text-textDark font-500 pt-5 mb-4">SUMMARY</h1>
+      <div className="w-full overflow-x-auto scrollbar-none custom-scrollbar">
+        <div className="flex justify-between">
+          {sortedNutrients
+            .filter((item) => item.nutrient.amount !== -9999)
+            .filter(
+              (item) =>
+                item.nutrient.name === "Carbohydrate, total" ||
+                item.nutrient.name === "Total fat" ||
+                item.nutrient.name === "Protein" ||
+                item.nutrient.name === "Sodium"
+            )
+            .map((item, idx) => (
+              <div key={idx} className="mr-[35px]">
+                <div className="mt-2 flex justify-center items-center flex-col">
+                  <div className="w-[50px] h-[50px]">
+                    <CircularProgressbarWithChildren
+                      strokeWidth={10}
+                      value={item.rdi?.amount ? Math.ceil(item.percentDaily as number) : 0}
```

Comment:
You calculation for percentDaily is different at different place. I would say create a function called `getPercentDaily(amount: number)` and use that everywhere